src/platform/backends/qemu/qemu_virtual_machine.h

```diff
@@ -54,9 +57,10 @@ class QemuVirtualMachine final : public VirtualMachine
     void on_error();
     void on_shutdown();
     VirtualMachine::State state;
-    const IPAddress ip;
+    std::unique_ptr<IPAddress> ip;
```

Comment:
A unique_ptr would do the job, but I think it's the semantics required map well to an std::experimental::optional type